libs/server-sdk/src/bindings/c/sdk.cpp

```diff
@@ -0,0 +1,415 @@
+    LD_ASSERT_NOT_NULL(event_name);
+
+    TO_SDK(sdk)->Track(*TO_CONTEXT(context), event_name);
+}
+
+LD_EXPORT(void)
+LDServerSDK_TrackMetric(LDServerSDK sdk,
+                        LDContext context,
+                        char const* event_name,
+                        double metric_value,
+                        LDValue value) {
+    LD_ASSERT_NOT_NULL(sdk);
+    LD_ASSERT_NOT_NULL(context);
+    LD_ASSERT_NOT_NULL(event_name);
+    LD_ASSERT_NOT_NULL(value);
+
+    Value* as_value = TO_VALUE(value);
+
+    TO_SDK(sdk)->Track(*TO_CONTEXT(context), event_name, metric_value,
+                       std::move(*as_value));
+
+    LDValue_Free(value);
+}
+
+LD_EXPORT(void)
+LDServerSDK_TrackData(LDServerSDK sdk,
+                      LDContext context,
+                      char const* event_name,
+                      LDValue value) {
+    LD_ASSERT_NOT_NULL(sdk);
+    LD_ASSERT_NOT_NULL(context);
+    LD_ASSERT_NOT_NULL(event_name);
+    LD_ASSERT_NOT_NULL(value);
+
+    Value* as_value = TO_VALUE(value);
+
+    TO_SDK(sdk)->Track(*TO_CONTEXT(context), event_name, std::move(*as_value));
+
+    LDValue_Free(value);
+}
+
+LD_EXPORT(void)
+LDServerSDK_Flush(LDServerSDK sdk, unsigned int reserved) {
+    LD_ASSERT_NOT_NULL(sdk);
+    LD_ASSERT(reserved == LD_NONBLOCKING);
+    TO_SDK(sdk)->FlushAsync();
+}
+
+LD_EXPORT(void)
+LDServerSDK_Identify(LDServerSDK sdk, LDContext context) {
+    LD_ASSERT_NOT_NULL(sdk);
+    LD_ASSERT_NOT_NULL(context);
+
+    TO_SDK(sdk)->Identify(*TO_CONTEXT(context));
+}
+
+LD_EXPORT(bool)
+LDServerSDK_BoolVariation(LDServerSDK sdk,
+                          LDContext context,
+                          char const* flag_key,
+                          bool default_value) {
+    LD_ASSERT_NOT_NULL(sdk);
+    LD_ASSERT_NOT_NULL(context);
+    LD_ASSERT_NOT_NULL(flag_key);
+
+    return TO_SDK(sdk)->BoolVariation(*TO_CONTEXT(context), flag_key,
+                                      default_value);
+}
+
+LD_EXPORT(bool)
+LDServerSDK_BoolVariationDetail(LDServerSDK sdk,
+                                LDContext context,
+                                char const* flag_key,
+                                bool default_value,
+                                LDEvalDetail* out_detail) {
+    LD_ASSERT_NOT_NULL(sdk);
+    LD_ASSERT_NOT_NULL(context);
+    LD_ASSERT_NOT_NULL(flag_key);
+
+    return MaybeDetail(sdk, out_detail, [&](Client* client) {
+        return client->BoolVariationDetail(*TO_CONTEXT(context), flag_key,
+                                           default_value);
+    });
+}
+
+LD_EXPORT(char*)
+LDServerSDK_StringVariation(LDServerSDK sdk,
+                            LDContext context,
+                            char const* flag_key,
+                            char const* default_value) {
+    LD_ASSERT_NOT_NULL(sdk);
+    LD_ASSERT_NOT_NULL(context);
+    LD_ASSERT_NOT_NULL(flag_key);
+    LD_ASSERT_NOT_NULL(default_value);
+
+    // TODO: custom allocation / free routines
```

Comment:
Is there an associated issue? Or out of consideration?

Reply:
Thanks for calling this out. I think it's somewhat pointless in this SDK, compared to the C SDK, because we don't have any support for custom allocators throughout the rest of the SDK.

I'll remove the comment. It is documented that users need to call `LDMemory_FreeString`.